task_processing/plugins/mesos/execution_framework.py

```diff
@@ -441,6 +596,126 @@ def create_new_docker_task(self, offer, task_config, available_ports):
+                type='COMMAND',
+                command=Dict(
+                    value='curl 127.0.0.1:{port}'.join(port=8888),
+                )
+            )
+        return task_info
+
+    def _extract_task_infos(self, task_operation, task_group_operations):
+        task_infos = []
+        if task_operation is not None:
+            task_infos = task_operation.launch.task_infos
+        if len(task_group_operations) > 0:
+            for task in task_group_operations:
+                task_infos = task_infos + task.launch_group.task_group.tasks
+
+        return task_infos
+
+    def _aggregate_status_updates_for_pod(self, pod_id, task_state):
+        sm = m()
+        with self._lock:
+            for task_id in self._pod_to_task_mappings[pod_id]:
+                sm.set(
+                    task_id,
+                    self.task_metadata[task_id].task_state_history
+                    [task_state]['status_update']
+                )
+
+        return sm
+
+    def _aggregate_task_configs_for_pod(self, pod_id):
+        tm = m()
+        with self._lock:
+            for task_id in self._pod_to_task_mappings[pod_id]:
+                tm.set(
+                    task_id,
+                    self.task_metadata[task_id].task_config
+                )
+
+        return tm
+
+    def _update_internal_state(self, task_id, task_state, metadata, update):
+        is_terminal = True if task_state in self._terminal_task_counts \
+            else False
+
+        with self._lock:
+            self.task_metadata = self.task_metadata.set(
+                task_id,
+                metadata.set(
+                    task_state=task_state,
+                    task_state_history=metadata.task_state_history.set(
+                        task_state,
+                        m(
+                            time_stamp=time.time(),
+                            status_update=update
+                        )
+                    ),
+                    terminated=is_terminal,
+                    terminal_raw_status_update=update,
+                )
+            )
+
+        if metadata.pod_id is not None:
+            # This container belongs to a pod. Iterate over all the nested
+            # tasks and check the status of those tasks.
+            all_tasks_terminated = True
+            state_changed = True
+            for t in self._pod_to_task_mappings[metadata.pod_id]:
+                with self._lock:
+                    md = self.task_metadata[t]
+                # Check if all the tasks in the pod have received a
+                # terminal status update or not.
+                if md.terminated is False:
+                    all_tasks_terminated = False
+
+                if task_state not in md.task_state_history:
+                    state_changed = False
+
+            if all_tasks_terminated is True or state_changed is True:
+                # This means that all the nested containers in the pod have
```

Comment:
the comment doesn't align with what is actually happening? you're doing this on a state change for *any* task in the pod afaict?